drake/systems/plants/collision/ResultCollector.h

```diff
@@ -0,0 +1,32 @@
+#ifndef DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+#define DRAKE_SYSTEMS_PLANTS_COLLISION_RESULTCOLLECTOR_H_
+
+#include "drake/systems/plants/collision/DrakeCollision.h"
+
+namespace DrakeCollision {
+Eigen::Vector3d toVector3d(const Eigen::Vector3d& vec);
+
+class ResultCollector {
+ public:
+  virtual ~ResultCollector() {}
+
+  virtual void addPointPairResult(const PointPair& result);
+
+  inline void addSingleResult(const ElementId idA, const ElementId idB,
+                              const Eigen::Vector3d& ptA,
+                              const Eigen::Vector3d& ptB,
+                              const Eigen::Vector3d& normal, double distance) {
+    addPointPairResult(PointPair(idA, idB, ptA, ptB, normal, distance));
+  }
+
+  std::vector<PointPair> getResults() const { return pts; }
+
+  PointPair minDistPoint();
```

Comment:
That would require making `addPointPairResult` `const`, which would be an API break. Is it worth breaking API for a branch that wasn't intended to do anything but rearrange code?